scipy/stats/tests/test_distributions.py

```diff
@@ -2092,6 +2092,62 @@ def test_logcdf_logsf(self):
         logsf = stats.invgauss.logsf(110, 1.05)
         assert_allclose(logsf, -56.1467092416426)
 
+    def test_pdf(self):
+        # regression tests for improvements introduces in gh-13665
+        # these tests are bases on Gyner & Smyth (2016)
+
+        # infinite mean corresponds to an inverse-chisquare pdf.
+        assert_allclose(
+            stats.invgauss.pdf([-1, 0, 1, 2, np.inf, np.nan], mu=np.inf),
+            [0., 0., 0.24197072, 0.10984782, 0., np.nan],
+            atol=1e-08
+        )
+        # extreme values have zero density for any finite mean value
+        assert_allclose(stats.invgauss.pdf([-1, 0, np.inf], mu=1), [0, 0, 0])
+
+    def test_ppf_sf(self):
+        p = [0.000001, 0.00001, 0.0001, 0.001, 0.01, 0.1, 0.5, 0.9, 0.99,
+             0.999, 0.9999, 0.99999, 0.999999]
+        p1 = stats.invgauss.cdf(stats.invgauss.ppf(p, mu=1), mu=1)
+        abs_error = np.abs(p - p1)
+        # five-number summary of the absolute error
+        fns = np.percentile(
+            abs_error, [0, 25, 50, np.mean(abs_error), 75, 100]
+        )
+        assert_allclose(fns, [0., 0., 8.13151629e-20, 0.,
+                        5.55111512e-17, 2.22044605e-16], atol=1e-08)
+
+        q = stats.invgauss.ppf(p, mu=1)
+        q1 = stats.invgauss.ppf(stats.invgauss.cdf(q, mu=1), mu=1)
+        rel_error = np.abs(q1 - q) / q
+        fns2 = np.percentile(
+            rel_error, [0, 25, 50, np.mean(rel_error), 75, 100]
+        )
+        assert_allclose(fns2, [0., 0., 1.17819649e-16, 0.,
+                        3.50412747e-16, 1.68867330e-11], atol=1e-08)
+
```

Comment:
The tests above check that the CDF and PPF are inverses of one another. It would be good to see a similar check for the ISF. Perhaps there is good testing of the accuracy of the CDF somewhere, but if not, we have not really tested that the CDF and PPF are correct for the inverse gausss distribution. Consider comparing the results of `ppf` and `isf` against ground truth values generated using statmod's implementation. Include the code that can be used to verify the values as a comment in the code.

Reply:
Never mind, this is unnecessary because it's already done by generic tests. Also, these tests calculate the error, then check that the error is close to some value that you calculated before, presumably? There is no need to do that. We would just check that the error is close to zero.

Reply:
These just mimic the tests done in the paper to ensure that the 5-number summary is done accurately compared to other implementations. Basically tests if the absolute and relative errors from computing CDF via the PPF are as small as possible.

Reply:
I see. I will check whether Boost achieves machine precision for roundtrip ppf-cdf with these inputs.

Reply:
Yes, Boost would pass all these tests (except it doesn't handle the special case of `mu='inf'`, which is a separate thing).